.github/workflows/swift-toolchain.yml

```diff
@@ -1134,15 +1148,19 @@ jobs:
           variant: sccache
 
       - uses: nttld/setup-ndk@v1
+        if: matrix.os == 'Android'
         id: setup-ndk
         with:
           ndk-version: ${{ inputs.ANDROID_NDK_VERSION }}
 
       - name: Configure zlib
         run: |
-          $NDKPATH = "${{ steps.setup-ndk.outputs.ndk-path }}"
-          if ( "${{ inputs.build_os }}" -eq "Windows" ) {
-            $NDKPATH = cygpath -m $NDKPATH
+          if ("${{ matrix.os }}" -eq "Android") {
+            $NDKPATH = "${{ steps.setup-ndk.outputs.ndk-path }}"
+            if ( "${{ inputs.build_os }}" -eq "Windows" ) {
+              $NDKPATH = cygpath -m $NDKPATH
+            }
+            $CMAKE_NDK_FLAG = "-DCMAKE_ANDROID_NDK=$NDKPATH"
```

Comment:
I think that this is why we were configuring the NDK unconditionally - it avoided this logic here. We would always pass along the NDKPATH and then rely on the `CMAKE_SYSTEM_NAME` to deal with the cross-compilation.

Reply:
I think we did it unconditionally because we were building only on win/x64 and there's an NDK for it. Since win/arm64 doesn't have one, this logic is necessary because passing an empty `CMAKE_ANDROID_NDK` value causes a failure.

Reply:
Might be worth having a comment about that.

Reply:
Done